test/Number/NumericSeparator.js

```diff
@@ -0,0 +1,227 @@
+        assert.throws(()=>eval('1_2____3'), SyntaxError, "Multiple numeric separators in a row are not allowed");
+        assert.throws(()=>eval('1_'), SyntaxError, "Decimal may not end in a numeric separator");
+        assert.throws(()=>eval('1__'), SyntaxError, "Decimal may not end in a numeric separator");
+        assert.throws(()=>eval('__1'), ReferenceError, "Decimal may not begin with a numeric separator");
+        assert.throws(()=>eval('_1'), ReferenceError, "Decimal may not begin with a numeric separator");
+
+        // Decimal with right-part with numeric separators
+        assert.throws(()=>eval('1.0__0'), SyntaxError, "Decimal right-part may not contain multiple contiguous numeric separators");
+        assert.throws(()=>eval('1.0_0__2'), SyntaxError, "Decimal right-part may not contain multiple contiguous numeric separators");
+        assert.throws(()=>eval('1._'), SyntaxError, "Decimal right-part may not be a single numeric separator");
+        assert.throws(()=>eval('1.__'), SyntaxError, "Decimal right-part may not be multiple numeric separators");
+        assert.throws(()=>eval('1._0'), SyntaxError, "Decimal right-part may not begin with a numeric separator");
+        assert.throws(()=>eval('1.__0'), SyntaxError, "Decimal right-part may not begin with a numeric separator");
+        assert.throws(()=>eval('1.0_'), SyntaxError, "Decimal right-part may not end with a numeric separator");
+        assert.throws(()=>eval('1.0__'), SyntaxError, "Decimal right-part may not end with a numeric separator");
+
+        // Decimal with both parts with numeric separators
+        assert.throws(()=>eval('1_.0'), SyntaxError, "Decimal left-part may not end in numeric separator");
+        assert.throws(()=>eval('1__.0'), SyntaxError, "Decimal left-part may not end in numeric separator");
+        assert.throws(()=>eval('1__2.0'), SyntaxError, "Decimal left-part may not contain multiple contiguous numeric separators");
+
+        // Decimal with exponent with numeric separators
+        assert.throws(()=>eval('1_e10'), SyntaxError, "Decimal left-part may not end in numeric separator");
+        assert.throws(()=>eval('1e_1'), SyntaxError, "Exponent may not begin with numeric separator");
+        assert.throws(()=>eval('1e__1'), SyntaxError, "Exponent may not begin with numeric separator");
+        assert.throws(()=>eval('1e1_'), SyntaxError, "Exponent may not end with numeric separator");
+        assert.throws(()=>eval('1e1__'), SyntaxError, "Exponent may not end with numeric separator");
+        assert.throws(()=>eval('1e1__2'), SyntaxError, "Exponent may not contain multiple contiguous numeric separators");
+
+        // Decimal big ints with numeric separators
+        assert.throws(()=>eval('1_n'), SyntaxError);
+    }
+  },
+  {
+    name: "Strings parsed as number do not support numeric separators for decimal literals",
+    body: function () {
+		assert.areEqual(NaN, Number('12_34'), "NaN === Number('12_34')");

```

Comment:
Nit: indentation.

Reply:
Thanks for noticing. There were tab characters on this line.